samza-core/src/main/java/org/apache/samza/runtime/LocalContainerRunner.java

```diff
@@ -71,14 +72,29 @@ public void run(StreamApplication streamApp) {
       Object taskFactory = TaskFactoryUtil.createTaskFactory(config, streamApp, this);
 
       SamzaContainer container = SamzaContainer$.MODULE$.apply(
-          containerModel.getProcessorId(),
           containerModel,
           config,
           jobModel.maxChangeLogStreamPartitions,
-          SamzaContainer.getLocalityManager(containerId, config),
           jmxServer,
           Util.<String, MetricsReporter>javaMapAsScalaMap(new HashMap<>()),
           taskFactory);
+      container.setContainerListener(
+          new SamzaContainerListener() {
+            @Override
+            public void onContainerStart() {
+              log.info("Container Started");
+            }
+
+            @Override
+            public void onContainerStop(boolean invokedExternally) {
+              log.info("Container Stopped");
+            }
+
+            @Override
+            public void onContainerFailed(Throwable t) {
+              log.info("Container Failed");
```

Comment:
Oh you are right. I think I actually wanted to ask you about this runner and I forgot about it. 

But why do we want to throw again, only to be caught by uncaughtexceptionhandler ? 

I am ok with throwing it after it returns from run. It will exit the status with error code I guess. Thanks for pointing out that I changed the existing behavior :D